dummy-api/matrix-service-api.yaml

```diff
@@ -0,0 +1,115 @@
+swagger: '2.0'
+info:
+  title: HCA Matrix Service
+  version: 0.0.0-dummy
+schemes:
+  - http
+  - https
+consumes:
+  - application/json
+produces:
+  - application/json
+securityDefinitions:
+  api_key:
+    type: apiKey
+    name: x-api-key
+    in: header
+definitions:
+  MatrixRequest:
+    type: object
+    properties:
+      bundle_fqids:
+        type: array
+        items:
+          type: string
+          format: dcp-fqid
+      bundle_fqids_url:
+        type: string
+        format: url
+      filter:
+        $ref: '#/definitions/MatrixFilter'
+      format:
+        $ref: '#/definitions/MatrixFormat'
+  MatrixFilter:
+    type: object
+    properties:
+      pandas_filter_string:
+        type: string
+  MatrixFormat:
+    type: string
+    enum:
+      - zarr
+      - parquet
+      - loom
+      - anndata
+      - singlecellexperiment
+  MatrixResponse:
+    type: object
+    properties:
+      request_id:
+        type: string
+        format: uuid
+      status:
+        $ref: '#/definitions/MatrixRequestStatus'
+      key:
+        type: string
+  MatrixRequestStatus:
+    type: string
+    enum:
+      - Initialized
+      - Running
+      - Complete
+      - Failed
+paths:
+  /matrix:
+    post:
+      security:
+        - api_key: []
+      x-amazon-apigateway-integration:
+        type: aws_proxy
+        httpMethod: POST
+        uri:
+          Fn::Join:
+            - ""
+            - - "arn:aws:apigateway:"
+              - Ref: "AWS::Region"
+              - ":lambda:path/2015-03-31/functions/"
+              - Fn::GetAtt: [MatrixPostFunction, Arn]
+              - "/invocations"
+      parameters:
+        - in: body
+          name: body
+          schema:
+            $ref: '#/definitions/MatrixRequest'
+      responses:
+        "201":
```

Comment:
Since this is an asynchronous operation, I would suggest a [202 (Accepted)](https://tools.ietf.org/html/rfc7231#section-6.3.3):

* [Azure Best Practices](https://docs.microsoft.com/en-us/azure/architecture/best-practices/api-design#asynchronous-operations)
* [RedHat learnings](https://restful-api-design.readthedocs.io/en/latest/methods.html#asynchronous-requests)

Reply:
Yep, will change.